### PR TITLE
Add bats test to measure goroutine stability

### PR DESCRIPTION
**Pull request by lack**
This test measures 100 container start/stop, and ensures the final total
of goroutines at the end does not exceed the goroutine count after
starting the first container.

Signed-off-by: Jim Ramsay <jramsay@redhat.com>

#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:

This adds some initial testing around goroutine stability; trying to ensure
that we don't have any common cases where the number of goroutines grow without
bounds.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by haircommander** [comment]
/ok-to-test
/approve

LGTM, thanks @lack

**Comment by TomSweeneyRedHat** [comment]
LGTM
and happy green test buttons.

**Review by saschagrunert — APPROVED** [review]
/lgtm

Thank you!

**Comment by haircommander** [comment]
hm the failures seem legit, though I can't find any evidence of why

/hold

can you maybe rebase again @lack?

**Comment by sohankunkerkar** [comment]
@cri-o/cri-o-maintainers I think it's good to cover this scenario in our test suite, thoughts?

**Comment by haircommander** [comment]
yeah agreed. @lack did you have bandwidth to continue with this or should we see if someone else wants to take it?

**Comment by lack** [comment]
Thanks! I had forgotten about this!

I don't have bandwidth to do much on it if changes are needed, but I don't mind rebase-and-rety if that's what's needed?

**Comment by sohankunkerkar** [comment]
@lack could you rebase this PR on top of the main branch?

**Pull request closed without merging**
